File: ocs_ci/templates/workloads/fsync.c
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <fcntl.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <stdio.h>
int main(int argc, char *argv[]){

        int fd, ret;
        while(1)
        {
                fd=open("/var/lib/www/html/mydir", O_DIRECTORY | O_RDONLY);
                if(fd < -1)
                {
                perror("creat()");

                }
                fsync(fd);
                close(fd);
        }
        return 9;

}

File: ocs_ci/templates/workloads/helper_scripts/fsync.py
import os


def fsync():
    while True:
        fd = os.open("/mnt/mydir/", os.O_DIRECTORY | os.O_RDONLY)
        print("type of fd: ", type(fd))
        if fd < -1:
            raise Exception("create")
        os.fsync(fd)
        os.close(fd)

    return 9


fsync()
